### DISCUSSION TITLE
The relay is switched off due to overheat that is not really happening.

### DISCUSSION THREAD

**Discussion by albertovn** [arendst/Tasmota#15598] [Q&A] [answered]
### PROBLEM DESCRIPTION
The ESP reports a too high temperature (right from the start it says 80ºC, so it must be off by more than 30ºC) triggering the overheat protection, switching off the relay immediately (now it says 95ºC, and as soon as I activate the relay the protection triggers and it switches off it). I would like to have a TempOffset command that is applied before the protection is triggered (this command has no effect right now in the temperature read for the overheat protection).

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [X] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [X] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [X] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [X] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [X] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [X] Device used (e.g., Sonoff Basic): Sonoff Dual R3
- [X] Tasmota binary firmware version number used: 11.1.0
  - [X] Pre-compiled
  - [ ] Self-compiled
- [X] Flashing tools used: Tasmota web installer
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
14:02:01.220 MQT: stat/tasmota_persiana_defectuoso/RESULT = {"NAME":"Sonoff Dual R3","GPIO":[32,0,0,0,0,0,0,0,0,576,225,0,0,0,0,0,0,0,0,0,0,7296,7328,224,0,0,0,0,160,161,0,0,0,0,0,0],"FLAG":0,"BASE":1}
14:02:01.446 MQT: stat/tasmota_persiana_defectuoso/RESULT = {"Module":{"0":"Sonoff Dual R3"}}
14:02:01.708 MQT: stat/tasmota_persiana_defectuoso/RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO6":{"0":"None"},"GPIO7":{"0":"None"},"GPIO8":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO11":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"576":"LedLink_i"},"GPIO14":{"225":"Relay2"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"0":"None"},"GPIO20":{"0":"None"},"GPIO21":{"0":"None"},"GPIO22":{"0":"None"},"GPIO23":{"0":"None"},"GPIO24":{"0":"None"},"GPIO25":{"7296":"CSE7761 Tx"},"GPIO26":{"7328":"CSE7761 Rx"},"GPIO27":{"224":"Relay1"},"GPIO32":{"160":"Switch1"},"GPIO33":{"161":"Switch2"},"GPIO34":{"0":"None"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO37":{"0":"None"},"GPIO38":{"0":"None"},"GPIO39":{"0":"None"}}

```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
14:02:47.502 MQT: stat/tasmota_persiana_defectuoso/STATUS = {"Status":{"Module":0,"DeviceName":"persiana_defectuoso","FriendlyName":["Persiana (Defectuoso)",""],"Topic":"tasmota_persiana_defectuoso","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":0,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
14:02:47.514 MQT: stat/tasmota_persiana_defectuoso/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota32/release/tasmota32.bin","RestartReason":"Software reset CPU","Uptime":"0T01:01:42","StartupUTC":"2022-05-11T12:01:05","Sleep":50,"CfgHolder":4617,"BootCount":8,"BCResetTime":"2022-04-01T10:30:37","SaveCount":39}}
14:02:47.524 MQT: stat/tasmota_persiana_defectuoso/STATUS2 = {"StatusFWR":{"Version":"11.1.0(tasmota)","BuildDateTime":"2022-04-13T06:40:58","Core":"2_0_3","SDK":"v.4.4.0","CpuFrequency":80,"Hardware":"ESP32-D0WD-V3","CR":"453/699"}}
14:02:47.537 MQT: stat/tasmota_persiana_defectuoso/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["VMF-WiFi-24",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["0000C009","2805C80001000600003C5A0A190000000000","40000080","00006000","00004000"]}}
14:02:47.567 MQT: stat/tasmota_persiana_defectuoso/STATUS4 = {"StatusMEM":{"ProgramSize":1394,"Free":1856,"Heap":145,"StackLowMark":3,"PsrMax":0,"PsrFree":0,"ProgramFlashSize":4096,"FlashSize":4096,"FlashFrequency":40,"FlashMode":3,"Features":["00000809","9FDAC7CF","0015A001","B7F7BFCF","05DA9BC0","E0360DC7","400840D2","20200000","0434482D"],"Drivers":"1,2,3,4,5,7,8,9,10,11,12,14,16,17,20,21,24,26,27,29,34,35,38,50,52,56,59,62,82,86","Sensors":"1,2,3,5,6,7,8,9,10,11,12,13,14,15,17,18,19,20,21,22,26,28,31,34,37,39,40,42,43,45,51,52,55,56,58,59,64,66,67,74,85,92,95,!127"}}
14:02:47.584 MQT: stat/tasmota_persiana_defectuoso/STATUS5 = {"StatusNET":{"Hostname":"tasmota-persiana-defectuoso-098","IPAddress":"192.168.1.60","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.3","DNSServer2":"80.58.61.250","Mac":"08:3A:F2:24:E3:DC","Ethernet":{"Hostname":"","IPAddress":"0.0.0.0","Gateway":"0.0.0.0","Subnetmask":"0.0.0.0","DNSServer1":"0.0.0.0","DNSServer2":"0.0.0.0","Mac":"00:00:00:00:00:00"},"Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
14:02:47.596 MQT: stat/tasmota_persiana_defectuoso/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.1.2","MqttPort":1883,"MqttClientMask":"persiana_defectuoso","MqttClient":"persiana_defectuoso","MqttUser":"vmf","MqttCount":2,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
14:02:47.610 MQT: stat/tasmota_persiana_defectuoso/STATUS7 = {"StatusTIM":{"UTC":"2022-05-11T13:02:47","Local":"2022-05-11T14:02:47","StartDST":"2022-03-27T02:00:00","EndDST":"2022-10-30T03:00:00","Timezone":"+01:00","Sunrise":"05:13","Sunset":"20:18"}}
14:02:47.618 MQT: stat/tasmota_persiana_defectuoso/STATUS9 = {"StatusPTH":{"PowerDelta":[0,0,0],"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0}}
14:02:47.642 MQT: stat/tasmota_persiana_defectuoso/STATUS10 = {"StatusSNS":{"Time":"2022-05-11T14:02:47","Switch1":"ON","Switch2":"ON","ENERGY":{"TotalStartTime":"2022-04-01T10:30:37","Total":0.001,"Yesterday":0.000,"Today":0.001,"Power":[ 0, 0],"ApparentPower":[ 0, 0],"ReactivePower":[ 0, 0],"Factor":[0.00,0.00],"Frequency": 0,"Voltage": 0,"Current":[0.000,0.000]},"Shutter1":{"Position":100,"Direction":0,"Target":100,"Tilt":0}}}
14:02:47.660 MQT: stat/tasmota_persiana_defectuoso/STATUS11 = {"StatusSTS":{"Time":"2022-05-11T14:02:47","Uptime":"0T01:01:42","UptimeSec":3702,"Heap":147,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":22,"MqttCount":2,"Berry":{"HeapUsed":3,"Objects":40},"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"VMF-WiFi-24","BSSId":"7A:83:C2:C1:46:D6","Channel":1,"Mode":"11n","RSSI":22,"Signal":-89,"LinkCount":1,"Downtime":"0T00:00:05"}}}
14:02:47.699 MQT: stat/tasmota_persiana_defectuoso/STATUS13 = {"StatusSHT":{"SHT0":{"Relay1":1,"Relay2":2,"Open":300,"Close":300,"50perc":50,"Delay":0,"Opt":"0000","Calib":[300,500,700,900,1000],"Mode":"0"}}}

```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
14:06:20.073 C61: Rx FFFFFFEC44
14:06:20.077 C61: F8950, U2255473, I0/0, P0/0
14:06:20.093 SRC: WebGui from 192.168.1.42
14:06:20.098 SRC: WebGui from 192.168.1.42
14:06:20.103 CMD: Grp 0, Cmd 'SHUTTERPOSITION', Idx 1, Len 9, Pld -99, Data 'StopClose'
14:06:20.111 SHT: Pos. in: payload StopClose (9), payload -99, idx 1, src 7
14:06:20.117 SHT: Pos. in: payload StopClose (0), payload 0, idx 1, src 7
14:06:20.122 SHT: lastsource 7:, real 587000, target 0, tiltreal: 0, tilttarget: 0, payload 0
14:06:20.128 SHT: Delay Start? var1 <99>=<>, max10s?
14:06:20.135 SRC: Shutter
14:06:20.139 SHT: Switched relay 2 by Shutter
14:06:20.155 MQT: stat/tasmota_persiana_defectuoso/RESULT = {"POWER2":"ON"}
14:06:20.168 MQT: stat/tasmota_persiana_defectuoso/POWER2 = ON
14:06:20.181 MQT: stat/tasmota_persiana_defectuoso/RESULT = {"ShutterPosition1":0}
14:06:20.234 C61: Tx A526
14:06:20.238 C61: Rx 2274108E
14:06:20.241 C61: Tx A523
14:06:20.245 C61: Rx 22F61F
14:06:20.248 C61: Tx A524
14:06:20.252 C61: Rx 0001C471
14:06:20.256 C61: Tx A52C
14:06:20.259 C61: Rx 00000334F7
14:06:20.263 C61: Tx A525
14:06:20.267 C61: Rx 0001D361
14:06:20.270 C61: Tx A52D
14:06:20.274 C61: Rx 0000082500
14:06:20.278 C61: F8950, U2257936, I0/0, P0/0
14:06:20.316 SHT: Shtr1 Real 585000, Start 587000, Stop 0, Dir -1, Delay 0, Rtc 0.15 [s], Freq 0, PWM 0, Tilt 0
14:06:20.329 MQT: stat/tasmota_persiana_defectuoso/RESULT = {"Shutter1":{"Position":98,"Direction":-1,"Target":0,"Tilt":0}}
14:06:20.340 SHT: Time 4(4), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 584000, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
14:06:20.413 SHT: Time 5(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 583000, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
14:06:20.433 C61: Tx A526
14:06:20.437 C61: Rx 2270643E
14:06:20.441 C61: Tx A523
14:06:20.444 C61: Rx 22F61F
14:06:20.448 C61: Tx A524
14:06:20.451 C61: Rx 0001C96C
14:06:20.455 C61: Tx A52C
14:06:20.459 C61: Rx 000007A483
14:06:20.462 SHT: Time 6(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 582000, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
14:06:20.472 C61: Tx A525
14:06:20.476 C61: Rx 0001CC68
14:06:20.479 C61: Tx A52D
14:06:20.483 C61: Rx FFFFFBFD37
14:06:20.487 C61: F8950, U2256996, I0/0, P0/0
14:06:20.512 SHT: Time 7(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 581000, nStop 0, Trgt 0, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
14:06:20.525 SHT: Shtr1 Real 581000, Start 587000, Stop 0, Dir -1, Delay 0, Rtc 0.35 [s], Freq 0, PWM 0, Tilt 0
14:06:20.542 MQT: stat/tasmota_persiana_defectuoso/RESULT = {"Shutter1":{"Position":97,"Direction":-1,"Target":0,"Tilt":0}}
14:06:20.554 NRG: Temperature 95.0
14:06:20.557 SRC: Overtemp
14:06:20.561 SHT: Switched relay 2 by Overtemp
14:06:20.565 SHT: Shtr1, Target 580000, Power: 0, tiltmv: 0
14:06:20.580 MQT: stat/tasmota_persiana_defectuoso/RESULT = {"POWER1":"OFF"}
14:06:20.593 MQT: stat/tasmota_persiana_defectuoso/POWER1 = OFF
14:06:20.604 MQT: stat/tasmota_persiana_defectuoso/RESULT = {"POWER2":"OFF"}
14:06:20.616 MQT: stat/tasmota_persiana_defectuoso/POWER2 = OFF
14:06:20.630 SHT: Time 9(1), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 579000, nStop 0, Trgt 580000, mVelo 0, Dir -1, Tilt 0, TrgtTilt: 0, Tiltmove: 0
14:06:20.640 C61: Tx A526
14:06:20.644 SHT: Stop 1 Mode 0 time 10
14:06:20.648 C61: Rx 2274F0AE

```

### TO REPRODUCE
_Steps to reproduce the behavior:_
When the reported temperature of the ESP is 'low' it works correctly. When is 'high' when I press the up or down button (either by pressing a physical button attached to the inputs, integrated button, web UI or Home Assistant) it starts working and stops right after less than 1 second, with time varying a bit. The temperature reported by the ESP is off by more than 30ºC, as just from the powering on it shows 80ºC, triggering the overheat protection too early.

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
We should have a way to correct the temperature reported by the ESP before the overheat protection is applied, as in this case the value is way off.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._
If I deactivate the sensor (SetSensor127 0) the issue disappears, so it can be considered a workaround, but obviously I stop receiving the temperature report in Home Assistant, which I would like to have (it also deactivates the overheating protection, and I would like to have that also in case it really overheats...).
Thanks!

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by albertovn** [comment]
Hi.
I'm not sure why my original issue was moved to this discussion (I think it's clearly an issue, the system is not prepared for cases where the reported ESP temperature is not accurate, and the fix would be as easy as set up a specific parameter to offset this reported temperature before the Overtemp condition is checked), but I have no problem discussing other approaches to fix this. The only thing I would hate is that this is is left unanswered and forgotten...
Please, any suggestions? any opinions on what to do? someone who thinks that this should be left as it is could give me reasons why?
Thanks!

**Comment by sfromis** [comment, marked as answer]
Well, to me this looks squarely as a question of how to use Tasmota, as you can config lots of options. You did not post where the temperature of 95 degrees came from. If it is the ESP32 temperature, you may want to consider turning off this entirely useless value using `SetSensor127 0`. You also have [SetOption42](https://tasmota.github.io/docs/Commands/#setoption42) to control the threshold.

**Reply by albertovn** [reply]
Hi,
OK, I wasn't aware of the SetOption42 to control the threshold, sorry. I will use that one. I don't agree with the term 'useless', as it's variation is correct, it's just the offset that is extremely high. Still I think it makes more sense to have an offset configurable value applied to the measure before the Overtemp condition is checked, but as long as I can at least control the threshold to keep the functionality alive I'm OK with it.
Thanks!

**Reply by sfromis** [reply]
Tasmota does have [TempOffset](https://tasmota.github.io/docs/Commands/#tempoffset), but that does by design not allow for extreme corrections of useless values.

**Reply by albertovn** [reply]
TempOffset is applied AFTER the Overtemp condition is checked, so in this case this configuration key is really useless. I think some other configuration key that is applied BEFORE the Overtemp condition is checked would be useful (and if it allows for extreme corrections that would render the useless value into useful, even better).

**Reply by sfromis** [reply]
Making rules based on crap values is a bad strategy. That's why I suggested turning off the noise.

**Reply by albertovn** [reply]
The point is that I don't agree is a crap value. The offset is unusually high, but the variation is good. If I have a way to compensate the offset the value would be totally good, so we can ignore a good information (knowing the ESP temperature I think is critical to avoid problems if something goes wrong, in the worst case scenario it could lead to fire that could be prevented if we can make the information useful), or we can have an extra tool to make this totally useful.
In any case, I don't want to argue forever, is clear that we disagree here. As long as I can tune the threshold using the SetOption42 you mentioned, I'm OK (not perfect, but useful).
